### PR TITLE
[stable/datadog] allow for setting annotations on cluster checks pod

### PR DESCRIPTION
**Pull request by jhohertz**
Signed-off-by: Joe Hohertz <joe@viafoura.com>

@hkaj 
@irabinovitch 
@CharlyF 
@mfpierre 
@clamoriniere 
@xlucas 

#### What this PR does / why we need it:

simply allows specifying podAnnotation for clusterChecks deployment, as is allowed for other components.

#### Which issue this PR fixes

none known

#### Special notes for your reviewer:

none

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by jhohertz** [comment]
Not sure what the linter is on about.... there was previously a space, but this has been addressed, and it is continuing to flag a now fixed problem.

**Comment by zanhsieh** [comment]
@jhohertz 
Would you mind to resolve chart conflict please? Thank you.
stable/datadog/Chart.yaml
stable/datadog/README.md

File: stable/datadog/values.yaml [hunk 1]
```diff
@@ -731,3 +731,9 @@ clusterchecksDeployment:
   # env:
   #   - name: <ENV_VAR_NAME>
   #     value: <ENV_VAR_VALUE>
+
+  ## @param podAnnotations - list of key:value strings - optional
+  ## Annotations to add to the cluster-agents's pod(s)
```
Comment (xlucas) [inline stable/datadog/values.yaml:837]:
```suggestion
  ## Annotations to add to pods of the Cluster Check Deployment.
```

File: stable/datadog/README.md [hunk 2]
```diff
@@ -336,6 +336,7 @@ helm install --name <RELEASE_NAME> \
...
-| `clusterchecksDeployment.rbac.dedicated`                  | If true, use dedicated RBAC resources for clusterchecks agent's pods                          | `false`                                     |
+| `clusterchecksDeployment.rbac.dedicated`                 | If true, use dedicated RBAC resources for clusterchecks agent's pods                          | `false`                                     |
 | `clusterchecksDeployment.rbac.serviceAccount`            | existing ServiceAccount to use (ignored if rbac.create=true) for clusterchecks                | `default`                                   |
 | `clusterchecksDeployment.strategy`                       | Which update strategy to deploy the Cluster Checks Deployment                                 | RollingUpdate with 0 maxUnavailable, 1 maxSurge |
+| `clusterchecksDeployment.podAnnotations`                 | Annotations to add to the Cluster Checks Pod(s)                                               | `nil`                                       |
```
Comment (xlucas) [inline stable/datadog/README.md]: Let rephrase this to `Annotations to add to pods of the Cluster Check Deployment`.

**Review by xlucas — CHANGES_REQUESTED** [review]

File: stable/datadog/values.yaml (hunk 1 above)
Comment (jhohertz) [reply stable/datadog/values.yaml:837]: done

File: stable/datadog/README.md (hunk 2 above)
Comment (jhohertz) [reply stable/datadog/README.md]: done

**Comment by jhohertz** [comment]
Rebased and addressed feedback

**Comment by xlucas** [comment]
Hi there, sorry for the delay, could you rebase again on master? I'll merge it right away when there's no more conflict.

**Comment by jhohertz** [comment]
Rebased
